### PR TITLE
V6/settings

### PR DESCRIPTION
**Pull request by gamalielhere**
### Devop

* \[x] Updated CHANGELOG.md
* \[x] Add PR label
* \[x] No hard-coded strings

### CODE REVIEW

File: src/core/store/global/getters.js [hunk 1]
```diff
@@ -17,7 +19,16 @@ const network = function (state) {
   }
   return network;
 };
+
+const gasPrice = function (state) {
```
Comment (kvhnuke) [inline src/core/store/global/getters.js:23]: gasPrice should always return wei amount, and the place where we use it we can switch it to gwei for display purpose

Comment (kvhnuke) [reply src/core/store/global/getters.js:23]: this func should just be `return getGasBasedOnType(state.baseGasPrice, state.gasPriceType)`

File: src/modules/send/ModuleSend.vue [hunk 2]
```diff
@@ -181,9 +181,9 @@ export default {
   },
   computed: {
     ...mapState('wallet', ['balance', 'web3', 'address']),
-    ...mapState('global', ['online', 'gasPrice']),
+    ...mapState('global', ['online']),
     ...mapState('external', ['ETHUSDValue']),
-    ...mapGetters('global', ['network']),
+    ...mapGetters('global', ['network', 'gasPrice']),
```
Comment (kvhnuke) [inline src/modules/send/ModuleSend.vue:186]: overhere we should get gasprice in wei

File: src/modules/settings/ModuleSettings.vue [hunk 3]
```diff
@@ -49,18 +55,25 @@
 <script>
 import ImportConfig from './components/SettingsImportConfig';
 import ExportConfig from './components/SettingsExportConfig';
-import Notifications from './components/SettingsNotification';
+// import Notifications from './components/SettingsNotification';
```
Comment (kvhnuke) [inline src/modules/settings/ModuleSettings.vue:58]: commented code

File: src/core/store/global/getters.js [hunk 4]
```diff
@@ -1,4 +1,6 @@
 import nodeList from '@/utils/networks';
+import { getGasBasedOnType } from '@/core/helpers/gasPriceHelper';
+import web3Utils from 'web3-utils';
```
Comment (kvhnuke) [inline src/core/store/global/getters.js:3]: `import {fromWei} from 'web3-utils'`

File: src/modules/settings/handler/handlerSettings.js [hunk 5]
```diff
@@ -0,0 +1,96 @@
+import vuexStore from '@/core/store';
+import { mapActions } from 'vuex';
+import { toWei, _ } from 'web3-utils';
+import xss from 'xss';
+import { Toast, ERROR } from '@/modules/toast/handler/handlerToast';
+
+const VALID_FIELDS = [
```
Comment (kvhnuke) [inline src/modules/settings/handler/handlerSettings.js:7]: shouldn't this be just the keys of the state?

File: src/modules/settings/handler/handlerSettings.js [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+  exportStore() {
+    const body = document.body;
+    const time = new Date();
+    const filename = `Store-Export-${time.getTime()}.json`;
+    const newObj = Object.assign({}, this.$store.state.global);
+    const stringifiedGasPrice = newObj.gasPrice.toString();
+    delete newObj['localStore'];
+    delete newObj['gasPrice'];
+    newObj['gasPrice'] = stringifiedGasPrice;
```
Comment (kvhnuke) [inline src/modules/settings/handler/handlerSettings.js:61]: I dont think this is necessary BN numbers will get converted to strings when you send through JSON.stringify also you dont have to delete localStore

**Review by kvhnuke — CHANGES_REQUESTED** [review]

**Review by gesseekur — APPROVED** [review]

File: src/views/layouts-wallet/TheSwapLayout.vue [hunk 7]
```diff
@@ -254,13 +254,12 @@ export default {
   },
   computed: {
     ...mapState('wallet', ['web3', 'address']),
-    ...mapState('global', ['gasPrice']),
-    ...mapGetters('global', ['network']),
+    ...mapGetters('global', ['network', 'gasPrice']),
     totalFees() {
-      return toBN(this.totalGasLimit).mul(toBN(this.gasPrice)).toString();
+      return toBN(this.totalGasLimit).mul(toBN(this.gasPriceGwei)).toString();
```
Comment (kvhnuke) [inline src/views/layouts-wallet/TheSwapLayout.vue:259]: This should gaslimit*gasprice (in wei not gwei)

**Review by kvhnuke — CHANGES_REQUESTED** [review]

**Review by kvhnuke — APPROVED** [review]

**Pull request merged**